direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java

```diff
@@ -116,13 +127,13 @@ public int getPartitionMaxNumBlobs() {
     return Optional.ofNullable(cfg.get(PARTITION_MAX_BLOBS))
         .map(Object::toString)
         .map(Integer::valueOf)
-        .orElse(1000);
+        .orElse(1_000);
   }
 
   public long getPartitionMaxTimeSpanMs() {
     return Optional.ofNullable(cfg.get(PARTITION_MAX_TIME_SPAN_MS))
         .map(Object::toString)
         .map(Long::valueOf)
-        .orElse(-1L);
+        .orElseGet(this::getLogRollInterval);
```

Comment:
This is definitely not a good default. This equals to limiting number of blobs in partition to number of parallel instances of replication controller. Agree that we can deduce default from log roll interval, but has to be significantly higher (e.g. x20).

Reply:
I don't think that's any better. Let's think about what we want to achieve here:

1) We want partition with the shortest time span possible, so can progress watermark faster.
2) We want to limit number of blobs in partition, because overhead of reading a single file is huge (especialy for small files, we've seen this when combining splits in Hadoop FileInputFormat).
3) We want to limit total size of the partition (because it would take a long time to process - this slows down parallel processing, because we may need to throttle other partitions). This is similar to 2).

What I'm unsure whether we should care about number of `Partitions` here... If not, it's better to have more smaller partitions, which runner can distribute. Flink should handle this well, Spark may have some trouble.

We're definitely missing settings for 3), in order to provide reasonable defaults (it may be safer to have a larger default time-span than?).

WDYT?

Reply:
I think that the timespan should be a last-resort limit, so setting it to higher number causes less harm than if it is too small. Setting it to either -1 or 20 * rollPeriod is fine. Setting it to lower value seems suboptimal.